File: src/bridge/CobolRuntime.js
var fs = require('fs');
var path = require('path');
var ClassDeclaration = require('../csharp/nodes/ClassDeclaration');

var RuntimeStructure = {
    CobolRuntime: new ClassDeclaration('CobolRuntime')
};

exports.CobolRuntimeBindings = RuntimeStructure;

exports.attachRuntime = function(source: string): string {
    var runtimeSource = fs.readFileSync(path.join(__dirname, '/CobolRuntime.cs')).toString();

    return [source, runtimeSource].join('\n');
};

File: src/bridge/CobolRuntime.cs
public static class CobolRuntime
{
	public static string ToCobolString(string str , int size) {
		return str.PadRight(size);
	}

	public static string ToCobolString(int num, int size, bool signed = false)
	{
		if (signed) {
			return (num >= 0? "+" : "") + num.ToString ("D" + size);
		} else {
			return num.ToString ("D" + size);
		}
	}
}